### ISSUE TITLE
Build wheels for C-Blosc

### ISSUE THREAD

**Issue by FrancescAlted** [Blosc/c-blosc#313]
We are in the process to support wheels *officially* for python-blosc, but we would like to build wheels also for C-Blosc, a pure C project. @thewtex we know that `cmake` is a C/C++ project that provide wheels (very convenient btw), but we are not sure how to do it.  It would be great if you can provide some pointers on how to produce wheels for non-Python libraries like C-Blosc.   Thanks in advance!

**Comment by esc** [comment]
THe `tbb` package does this too: https://pypi.org/project/tbb/

**Comment by kalvdans** [comment]
Can't the python-blosc wheel statically link C-Blosc?

**Comment by esc** [comment]
@kalvdans this is what we have done for the last decade or so, yes. But this is about pure-c wheels for c-blosc. I imagine to allow other python packages to dynamically link against it?

**Comment by kalvdans** [comment]
Thanks for the clarification, @esc. I can only speak for h5py which would have no use for a c-blosc wheel because it goes through libhdf5.

**Comment by FrancescAlted** [comment]
> THe `tbb` package does this too: https://pypi.org/project/tbb/

Yes, Intel is doing that for several libraries, but unfortunately we don't have access to the sources :-/

**Comment by thewtex** [comment]
> We are in the process to support wheels officially for python-blosc

Excellent! Note that the existing configuration could easily be migrated to GitHub Actions for better GitHub integration and more build system support.

Also, in addition to the existing `dockcross/manylinux1-x64` builds, they could be updated to `dockcross/manylinux2014-x64`, which has newer compilers that provide better optimizations.

Additionally, it would be interesting to try cross-compiling ARM wheels with the `dockcross/manylinux2014-aarch64` images -- the same commands can be used. This image is more experimental, but it may be an easy way to provide ARM wheels.

> how to produce wheels for non-Python libraries like C-Blosc

Unfortunately, wheels do not provide standards for providing cross-platform loading of shared libraries and sharing of build artifacts, like CMake package configuration files or headers. While it is hard to provide a wheel that does this, conda packages do provide this, so a well-supported c-blosc package is helpful in that respect.

**Comment by FrancescAlted** [comment]
> > We are in the process to support wheels officially for python-blosc
> 
> Excellent! Note that the existing configuration could easily be migrated to GitHub Actions for better GitHub integration and more build system support.

Yes, we have just migrated to GitHub actions and cibuildwheel.  So far GHA is giving us plenty of speed and robustness, and cibuildwheel is offering a lot of flexibility for creating the wheels, so we are very happy so far.  Perhaps the only 'weak' point is that it is not easy to make scikitbuild with cibuildwheel, but we have configured the CMakeLists.txt files to use production defaults, and this is what it is used for making the wheels; it works pretty good in practice.  

> 
> Also, in addition to the existing `dockcross/manylinux1-x64` builds, they could be updated to `dockcross/manylinux2014-x64`, which has newer compilers that provide better optimizations.

Well, it looks like and we are providing `manylinux1` and `manylinux2010` for both x86 64 and 32 bit platforms.  Producing `manylinux2014` wheels is also a good idea.

> 
> Additionally, it would be interesting to try cross-compiling ARM wheels with the `dockcross/manylinux2014-aarch64` images -- the same commands can be used. This image is more experimental, but it may be an easy way to provide ARM wheels.

Yup.  Supporting ARM is becoming more and more important (specially after the lastest Apple move for Macs).

Anyway, thanks for triggering the initial wheel support for python-blosc!  Because of this we are now very close to make an official release of wheels for forthcoming python-blosc 1.10.0.
